### PR TITLE
fix savegame bug, cosmetic changes

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: src/p_saveg.c [hunk 1]
```diff
@@ -141,7 +141,7 @@ static void saveg_write_pad(void)
 
     for (i=0; i<padding; ++i)
     {
-        saveg_read8();
+        saveg_write8(0);
```
Comment (fabiangreffrath) [inline src/p_saveg.c:144]: Oh, wow! Won't this break savegame compatibility?

Comment (rfomin) [reply src/p_saveg.c:144]: No, the main point is to move the pointer to a `padding` number of bytes. If it's a `0` entry or garbage, it doesn't matter.

File: src/p_saveg.c [hunk 2]
```diff
@@ -1725,12 +1725,12 @@ static void saveg_read_scroll_t(scroll_t *str)
...
 
-    // fixed_t vdy;
-    str->vdy = saveg_read32();
-
     // fixed_t vdx;
     str->vdx = saveg_read32();
 
+    // fixed_t vdy;
+    str->vdy = saveg_read32();
+
```
Comment (rfomin) [inline src/p_saveg.c:1733]: This bug may break some saves. Try to save at the beginning of MAP02 AD_MORTEM.wad, enemies don't spawn later. Still compatible, no crashes.

File: src/p_saveg.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/p_saveg.c:144]: Only advances the `save_p` pointer, I get it. 👍

File: src/p_saveg.c (hunk 2 above)
Comment (fabiangreffrath) [reply src/p_saveg.c:1733]: I can live with that, no reason for a savegame version bump.

**Pull request merged**